Automattic/woocommerce-payments
# Fix: Handle payment methods errors in backend

Fixes #9177

#### Changes proposed in this Pull Request

* Whenever an error occurs while submitting a request to `v1/payment_methods`, instead of showing the error right away to the shopper, we'll submit the error to the backend for it to be recorded in a failed order.

I also had some issues with the Playwright tests being flaky, so I increased the timeouts to obtain more consistently successful builds.

#### Testing instructions





* ⚠️ **PREREQUISITE**: This requires a WooPayments account in Live mode, this won't work in Sandbox/Test mode.
* Add any product to the shopping cart, then go to blocks checkout (**Repeat for classic checkout afterwards**).
* Fill all the required information, for the credit card, use one of [Stripe test cards](https://docs.stripe.com/testing#cards)

Actual Behavior (without this PR):
* An error will be displayed to the shopper.
* As a merchant, if you login to wp-admin and then go to WooCommerce -> Orders, you won't see any new records for the previously failed attempt.

Expected behavior (with this PR):
* An error will be displayed to the shopper (this time it'll take longer since a request is being made to the backend).
* As a merchant, if you login to wp-admin and then go to WooCommerce -> Orders, you should see a Failed order for the previous attempt. 
* In the order notes, verify that the error message is also recorded.
![image](https://github.com/user-attachments/assets/0794732a-3edb-4ea9-9790-198aa1c0fb0e)

-------------------


**Post merge**
